### DISCUSSION TITLE
pogo vs trek

### DISCUSSION THREAD

**Discussion by ofer-pd** [NRWLDev/pogo-migrate#30] [Q&A]
First off, thanks for creating pogo.  I was just starting to learn [yoyo](https://pypi.org/project/yoyo-migrations/) when I noticed it doesn't support asyncpg, so then I searched for "asyncpg database migrations" and found pogo - as well as one other still-active asyncpg-based migration library:

https://pypi.org/project/asyncpg-trek/

Do you know about trek, and if so, how would you describe the differences?

It would also be great to pull @adriangb into the discussion to ask the same from their POV - especially since Discussions are not currently enabled in the trek project.

**Comment by adriangb** [comment]
I'm not sure what the differences are as I've never used pogo. I will say I don't really have time to do much OSS maintenance nowadays (working at Pydantic doing OSS maintenance is a full time job) so I can't promise much with `ayncpg-trek`.

**Reply by ofer-pd** [reply]
Ok.  Thanks for the update.

I'm going to try out pogo for now and see how that goes.

**Reply by EdgyEdgemond** [reply]
@adriangb Definitely focus on pydantic :D Thats way more useful.

@ofer-pd I was unaware of trek until this discussion, I will check it out. If theres anything groundbreaking or cool over there I'll take a look at adopting some of it.

Please do give pogo a try and raise issues if you find anything doesn't work or theres something you wish you could do with it, or if anything just feels unfriendly usability wise.

It's in use in quite a few work projects without any issues for me, so best of luck.

**Reply by EdgyEdgemond** [reply]
It looks like trek is designed for more flexibility in usability, pogo is aimed at basic cli usage. So local and docker based usage like trek should be relatively similar.

If you are interested in the use from an admin API etc, that would require custom code server side. But the code can all be imported so wouldn't be impossible with pogo.

In regards to testing, check https://github.com/NRWLDev/pytest-pogo, which provides some fixtures for getting setup in tests.
https://nrwldev.github.io/pogo-migrate/testing/

**Reply by EdgyEdgemond** [reply]
To answer the other question in your initial comment, it looks like pogo and trek are both designed with raw sql at heart. So functionally they appear to be the same. Implementation and features my differ but at the core looks like they were both built for the same purpose. With support for py migrations when you want to get real messy with a data migration.
